### PR TITLE
Material suffix

### LINKED ISSUE #27440: Enable the utilization of suffix for functor materials
### Motivation

The `prop_getter_suffix` parameter is not used in MOOSE right now and it would make the declaration of material properties for different phases of the same material easier.

### Design

Just add the suffix to the functor material names.

### Impact

Simpler syntax for functor materials for different phases of the same material.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #27440

### CODE REVIEW

File: framework/include/functormaterials/FunctorMaterial.h [hunk 1]
```diff
@@ -80,5 +80,10 @@ FunctorMaterial::addFunctorPropertyByBlocks(const std::string & name,
 
   _supplied_functor_props.insert(name);
   return _subproblem.addPiecewiseByBlockLambdaFunctor<T>(
-      prop_name, my_lammy, clearance_schedule, _mesh, sub_ids, _tid);
+      (_get_suffix == "" ? "" : _get_suffix + "_") + prop_name,
```
Comment (GiudGiud) [inline framework/include/functormaterials/FunctorMaterial.h:83]: this is a prefix not a suffix

File: test/tests/functormaterials/with_suffix/tests [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    type = CSVDiff
+    input = functor_material.i
+    csvdiff = functor_material_out.csv
+    requirement = 'The system should be able to add a generic functor material without a suffix for the properties.'
+  []
+  [suffix]
+    type = CSVDiff
+    input = functor_material.i
+    csvdiff = functor_material_out.csv
+    requirement = 'The system should be able to add a generic functor material with a suffix for the properties.'
```
Comment (GiudGiud) [inline test/tests/functormaterials/with_suffix/tests:14]:
```suggestion
    requirement = 'The system should be able to add a generic functor material with a prefix for the names of the properties.'
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]

File: framework/include/functormaterials/FunctorMaterial.h (hunk 1 above)
Comment (grmnptr) [reply framework/include/functormaterials/FunctorMaterial.h:83]: ahh nice

**Review by GiudGiud — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

**Comment by grmnptr** [comment]
Failures were unrelated.

**Pull request merged**